File: pkg/pwrgrep/testdata/fixtures/go/concurrency.go
package fixture

import "sync"

func f(mu *sync.Mutex) {
	// ruleid: go-concurrency
	done := make(chan bool)
	// ruleid: go-concurrency
	go work(done)
	// ruleid: go-concurrency
	select {
	case <-done:
	}
	// ruleid: go-concurrency
	mu.Lock()
	// ok: go-concurrency
	buf := make([]byte, 10)
	_ = buf
}

func work(done chan bool) { done <- true }

File: pkg/pwrgrep/testdata/fixtures/c/concurrency.c
#include <pthread.h>
#include <unistd.h>

static pthread_mutex_t lock;

void f(void *(*work)(void *)) {
    pthread_t t;
    // ruleid: c-concurrency
    pthread_create(&t, NULL, work, NULL);
    // ruleid: c-concurrency
    pthread_mutex_lock(&lock);
    // ruleid: c-concurrency
    pid_t p = fork();
    // ok: c-concurrency
    sleep(1);
}
